src/Features/CSharp/Portable/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -30,18 +32,31 @@ internal class CSharpAsAndNullCheckDiagnosticAnalyzer : AbstractCodeStyleDiagnos
 
         public CSharpAsAndNullCheckDiagnosticAnalyzer()
             : base(IDEDiagnosticIds.InlineAsTypeCheckId,
-                   new LocalizableResourceString(
-                       nameof(FeaturesResources.Use_pattern_matching), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
+                new LocalizableResourceString(
+                    nameof(FeaturesResources.Use_pattern_matching), FeaturesResources.ResourceManager, typeof(FeaturesResources)))
         {
         }
 
         protected override void InitializeWorker(AnalysisContext context)
-            => context.RegisterSyntaxNodeAction(SyntaxNodeAction, SyntaxKind.IfStatement);
+            => context.RegisterSyntaxNodeAction(SyntaxNodeAction,
+                SyntaxKind.IfStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.ReturnStatement,
+                SyntaxKind.LocalDeclarationStatement);
 
         private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
         {
+            var node = syntaxContext.Node;
+            var syntaxTree = node.SyntaxTree;
+
+            // "x is Type y" is only available in C# 7.0 and above. Don't offer this refactoring
+            // in projects targeting a lesser version.
+            if (((CSharpParseOptions)syntaxTree.Options).LanguageVersion < LanguageVersion.CSharp7)
```

Comment:
Tagging @mavasani .  We have to do this sort of a check in a lot of our analyzers.  It seems like it would be nice to avoid having to do all the callbacks for a file if we could do an up front check only once.